src/scripts/saveToOneNote/saveToOneNote.ts

```diff
@@ -179,19 +180,45 @@ export class SaveToOneNote {
 
 						Promise.all([getRevisionsPromise, timeoutPromise]).then((values) => {
 							let revisions = values[0] as OneNoteApi.Revision[];
-							this.getApi().updatePage(pageId, revisions).then(() => {
+
+							let thenCb = () => {
```

Comment:
To avoid repeat code I pulled out the callbacks, but this still feels messy. I'm open to improvements if any of you have suggestions :)